compiler/ballerina-lang/src/main/java/io/ballerina/projects/DocumentContext.java

```diff
@@ -95,15 +94,7 @@ TextDocument textDocument() {
             return this.textDocument;
         }
 
-        // TODO: The content should be loaded from a TextLoader
-        Path documentPath = Paths.get(documentId().documentPath());
-        try {
-            String text = new String(Files.readAllBytes(documentPath), StandardCharsets.UTF_8);
-            this.textDocument = TextDocuments.from(text);
-        } catch (IOException e) {
-            // TODO improve error handling
-            throw new RuntimeException("Unable to read file: " + documentPath);
-        }
+        this.textDocument = TextDocuments.from(this.content);
```

Comment:
Let's assign only if the textDocument is null and then we can use only one return,  since we use the same twice now

Reply:
Fixed in 0d89c8f